avm/res/container-service/managed-cluster/main.bicep

```diff
@@ -0,0 +1,1012 @@
+param podIdentityProfileEnable bool = false
+
+@description('Optional. The pod identities to use in the cluster.')
+param podIdentityProfileUserAssignedIdentities array?
+
+@description('Optional. The pod identity exceptions to allow.')
+param podIdentityProfileUserAssignedIdentityExceptions array?
+
+@description('Optional. Whether the The OIDC issuer profile of the Managed Cluster is enabled.')
+param enableOidcIssuerProfile bool = false
+
+@description('Optional. Whether to enable Workload Identity. Requires OIDC issuer profile to be enabled.')
+param enableWorkloadIdentity bool = false
+
+@description('Optional. Whether to enable Azure Defender.')
+param enableAzureDefender bool = false
+
+@description('Optional. Whether to enable Kubernetes pod security policy. Requires enabling the pod security policy feature flag on the subscription.')
+param enablePodSecurityPolicy bool = false
+
+@description('Optional. Whether the AzureBlob CSI Driver for the storage profile is enabled.')
+param enableStorageProfileBlobCSIDriver bool = false
+
+@description('Optional. Whether the AzureDisk CSI Driver for the storage profile is enabled.')
+param enableStorageProfileDiskCSIDriver bool = false
+
+@description('Optional. Whether the AzureFile CSI Driver for the storage profile is enabled.')
+param enableStorageProfileFileCSIDriver bool = false
+
+@description('Optional. Whether the snapshot controller for the storage profile is enabled.')
+param enableStorageProfileSnapshotController bool = false
+
+@allowed([
+  'AKSLongTermSupport'
+  'KubernetesOfficial'
+])
+@description('Optional. The support plan for the Managed Cluster.')
+param supportPlan string = 'KubernetesOfficial'
+
+@description('Optional. The diagnostic settings of the service.')
+param diagnosticSettings diagnosticSettingType
+
+@description('Optional. Specifies whether the OMS agent is enabled.')
+param omsAgentEnabled bool = true
+
+@description('Optional. Resource ID of the monitoring log analytics workspace.')
+param monitoringWorkspaceId string?
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = true
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments roleAssignmentType
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. The resource ID of the disc encryption set to apply to the cluster. For security reasons, this value should be provided.')
+param diskEncryptionSetID string?
```

Comment:
Should be `diskEncryptionSetResourceId`. Not adding a suggestion as it must be changed in multiple locations.